apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server_test.py

```diff
@@ -0,0 +1,456 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import OrderedDict
+import common, datetime, json, os, urllib2
+
+# These tests can be run with pytest: http://doc.pytest.org/en/latest/index.html
+
+try:
+  server_address = 'http://' + os.environ[common.host_environment_variable]
+  request = urllib2.Request(server_address,
+                            headers={common.metadata_flavor : common.metadata_flavor_google})
+  urllib2.urlopen(request)
+except KeyError:
+  raise Exception('%s must be set' % common.host_environment_variable)
+except urllib2.URLError:
+  raise Exception('The emulator cannot be reached at %s' % server_address)
+
+def check_header(headers, url, key, value):
+  key_lower = key.lower()
+  assert key_lower in headers
+  assert headers[key_lower] == value
+
+def expect_content_absolute(url, expected, expected_content_type, expect_metadata_header=True):
+  request = urllib2.Request(url, headers={common.metadata_flavor : common.metadata_flavor_google})
+  response = urllib2.urlopen(request)
+  headers = response.info().dict
+
+  if expect_metadata_header:
+    check_header(headers, url, common.metadata_flavor, common.metadata_flavor_google)
+  else:
+    assert common.metadata_flavor.lower() not in headers
+
+  check_header(headers, url, common.server, common.server_value)
+  check_header(headers, url, common.content_type, expected_content_type)
+  
+  contents = response.read()
+  assert expected == contents
+
+def expect_content(path, expected, expected_content_type, expect_metadata_header=True):
+  expect_content_absolute(server_address + '/computeMetadata/v1/' + path,
+                          expected,
+                          expected_content_type,
+                          expect_metadata_header)
+
+def expect_error(path, expected_code, metadata_flavor_value=common.metadata_flavor_google):
+  url = server_address + '/computeMetadata/v1/' + path
```

Comment:
given you reference this path in a couple places, could make it a constant template about, and interpolate with .format() where needed